### ISSUE TITLE
Crash Report: Brave crashes in Catalina when checking for updates - 100% reproducible

### ISSUE THREAD

**Issue by Macaroon23** [brave/brave-browser#38600]
IMPORTANT: Your crash has already been automatically reported to our crash system. Please file this bug only if you can provide more information about it.

Brave Version: 1.66.113 Chromium: 125.0.6422.76
Operating System: Mac OS X 10.15.7

URL (if applicable) where crash occurred: 
About Brave

Can you reproduce this crash? 
Yes - 100% reproducibility with one simple step.

What steps will reproduce this crash? (If it's not reproducible, what were you doing just before the crash?)
1. Launch Brave
2. Crashes immediately on first launch
3. Relaunch Brave - does not crash if relaunched within a few minutes 
4. Brave functions normally until automatic or manual update check
5. Check for updates by opening "About Brave"
6. Instant and 100% reproducible crash
7. Open pages can be restored
8. Additional information:
9. It is highly likely that Brave updater is causing the crash in Catalina
10. Brave v1.66.110 *introduced* this bug, which was not fixed in v1.66.113
11. Brave v1.66.110 could only be updated to v1.66.113 by downloading v1.66.113 from Brave website and installing in place of v1.66.110
****DO NOT CHANGE BELOW THIS LINE****
Crash ID: crash/dd5c1200-293f-040c-0000-000000000000

**Comment by Macaroon23** [comment]
I have created a new, empty profile - crash behaviour is unchanged.

The bug is in Brave update check, not any third party page.

**Comment by Macaroon23** [comment]
@iefremov this crash report was for Brave for Mac v1.66.113

The issue persists in v1.66.115 however. Crash on launch is less common, but reloading 'About Brave' to force update check still causes 100% reproducible crashes. See crash logs below.

FWIW I note that Vivaldi browser update has fixed a bug that caused crashes when "Vivaldi" was typed in the address bar.

Hope this info helps.

Crash from Thursday 30 May 2024 at 16:11:36
Status:	Uploaded
Uploaded crash report ID:	c7700400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 16:11:38
Local crash context:	4c678d59-6c9b-4900-a327-70f5d27a40bb

Crash from Thursday 30 May 2024 at 14:57:21
Status:	Uploaded
Uploaded crash report ID:	ad210400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:58:21
Local crash context:	70bfba7a-4f46-48ec-b163-f7e3fcb140d4

Crash from Thursday 30 May 2024 at 14:56:36
Status:	Uploaded
Uploaded crash report ID:	e4200400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:57:22
Local crash context:	28d3b878-4a3b-4547-a847-7c4cad39e55f

Crash from Wednesday 29 May 2024 at 12:04:27
Status:	Uploaded
Uploaded crash report ID:	d51f0400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:56:17
Local crash context:	933443c2-7927-4a64-be22-8e9de80b68ed

Crash from Wednesday 29 May 2024 at 12:04:16
Status:	Uploaded
Uploaded crash report ID:	d81f0400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:56:18
Local crash context:	72e32f94-4a3c-4c75-aac8-ebe66567f175

Crash from Wednesday 29 May 2024 at 12:04:07
Status:	Uploaded
Uploaded crash report ID:	df1f0400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:56:19
Local crash context:	cd26f9c3-c10a-48b0-8dc4-59e3bbee5106

Crash from Wednesday 29 May 2024 at 12:03:55
Status:	Uploaded
Uploaded crash report ID:	d11f0400-c07b-0d0c-0000-000000000000
Upload time:	Thursday 30 May 2024 at 14:56:16
Local crash context:	4f98671c-468c-465f-b3b8-863f07ac068b

Crash from Wednesday 29 May 2024 at 12:03:31
Status:	Uploaded
Uploaded crash report ID:	7bc31a00-293f-040c-0000-000000000000
Upload time:	Wednesday 29 May 2024 at 12:03:57
Local crash context:	805706c4-1da6-4fa2-9313-a63cf96a6af9

Crash from Wednesday 29 May 2024 at 12:03:03
Status:	Uploaded
Uploaded crash report ID:	c7c21a00-293f-040c-0000-000000000000
Upload time:	Wednesday 29 May 2024 at 12:03:04
Local crash context:	cb93303d-471d-4e33-8d0d-a2f7111ca4a4

**Comment by Macaroon23** [comment]
New random crash.

Crash from Friday 31 May 2024 at 13:55:27</h3>
Status: | Uploaded
-- | --
Uploaded crash report ID: | 308b0800-c07b-0d0c-0000-000000000000
Upload time: | Friday 31 May 2024 at 13:55:28
Local crash context: | 6b5a0614-5caf-4dab-956b-e7a7b6bf1a18

**Comment by iefremov** [comment]
@Macaroon23 does it persist? we've fixed the macos updater recently

**Comment by Macaroon23** [comment]
Yes it persists. However the cause was identified by @atuchin-m in another thread.
It's network filters - in my case Lulu - that Brave doesn't play nicely with.

Disabling Lulu stopped the crashing behaviour on launch and when manually checking  _About Brave_.
Re-enabling Lulu brings back the crashes. I to the latest version of Lulu (2.6.3) but the crashes persist.

I guess the only solution is to temporarily disable Lulu when launching Brave, except... well, with Brave 128.1.69.168 I believe I'm running the final version for Catalina anyway, so there's no point in checking for updates anyway. 

The only solution now is to disable autoupdates on my Catalina Mac.

**Comment by bsclifton** [comment]
Hi - sorry for lack of update @Macaroon23. Does this still occur? I believe we have a known crash for Catalina already - not specific to updates. https://github.com/brave/brave-browser/issues/39101

@iefremov do you know if this is a duplicate of that one?
